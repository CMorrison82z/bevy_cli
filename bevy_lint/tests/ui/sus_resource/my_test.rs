#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::events_resource_system_param)]

use bevy::prelude::*;

fn main() {
    App::new().add_systems(Startup, (bad_system,)).run();
}

//~| HELP: events Resource footgun
//~v ERROR: use of bad system parameter
fn bad_system(_resource_events: Res<Events<AppExit>>) {}
