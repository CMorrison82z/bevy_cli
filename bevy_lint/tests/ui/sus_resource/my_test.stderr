error: use of bad system parameter
  --> tests/ui/sus_resource/my_test.rs:13:37
   |
13 | fn bad_system(_resource_events: Res<Events<AppExit>>) {}
   |                                     ^^^^^^^^^^^^^^^
   |
   = help: events Resource footgun
note: the lint level is defined here
  --> tests/ui/sus_resource/my_test.rs:3:9
   |
3  | #![deny(bevy::events_resource_system_param)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

